borsh/assembly/serializer.ts

```diff
@@ -48,7 +48,7 @@ export class BorshSerializer extends Serializer<ArrayBuffer> {
   }  */
     if (t != null) {
       this.buffer.store<u8>(1);
-      this.encode(<NonNullable<T>>t);
```

Comment:
Should we add a test in which this breaks?

Reply:
If you can find one!